File: Components/WinForms/VB/VertGridTutorials/HitInfo/HitInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data

Namespace DevExpress.XtraVerticalGrid.Demos.Tutorials
	''' <summary>
	''' Summary description for HitInfo.
	''' </summary>
	Public Partial Class HitInfo
		Inherits TutorialControl
		Public Sub New()
			'
			' Required for Windows Form Designer support
			'
			InitializeComponent()
			'
			' TODO: Add any constructor code after InitializeComponent call
			'
		End Sub
        Public Overrides ReadOnly Property SetNewWhatsThisPadding() As Boolean
            Get
                Return True
            End Get
        End Property
		Private Sub HitInfo_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
            DevExpress.XtraVerticalGrid.Design.XViews.ConfigureDemoView(vGridControl1)
		End Sub

		'<vGridControl1>
		Private Sub vGridControl1_MouseMove(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles vGridControl1.MouseMove
			ShowHitInfo(vGridControl1.CalcHitInfo(New Point(e.X, e.Y)))
		End Sub

		Private Sub ShowHitInfo(ByVal hi As DevExpress.XtraVerticalGrid.VGridHitInfo)
			lbHitTest.Text = hi.HitInfoType.ToString()
			If hi.Row Is Nothing Then
				lbRow.Text = "No row"
			Else
				If TypeOf hi.Row Is DevExpress.XtraVerticalGrid.Rows.MultiEditorRow Then
					If hi.CellIndex <> -1 Then
						lbRow.Text = hi.Row.GetRowProperties(hi.CellIndex).Caption
					End If
				Else
					lbRow.Text = hi.Row.Properties.Caption
				End If
			End If
			If hi.HitInfoType = DevExpress.XtraVerticalGrid.HitInfoTypeEnum.ValueCell Then
				If TypeOf hi.Row Is DevExpress.XtraVerticalGrid.Rows.MultiEditorRow Then
					lbCellValue.Text = vGridControl1.GetCellValue(TryCast(hi.Row, DevExpress.XtraVerticalGrid.Rows.MultiEditorRow), hi.RecordIndex, hi.CellIndex).ToString()
				Else
					lbCellValue.Text = vGridControl1.GetCellValue(hi.Row, hi.RecordIndex).ToString()
				End If
			Else
				lbCellValue.Text = " "
			End If
		End Sub
		'</vGridControl1>
	End Class
End Namespace

File: Components/WinForms/CS/VertGridTutorials/HitInfo/HitInfo.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

namespace DevExpress.XtraVerticalGrid.Demos.Tutorials {
    /// <summary>
    /// Summary description for HitInfo.
    /// </summary>
    public partial class HitInfo : TutorialControl {
        public HitInfo() {
            //
            // Required for Windows Form Designer support
            //
            InitializeComponent();
            //
            // TODO: Add any constructor code after InitializeComponent call
            //
        }
        public override bool SetNewWhatsThisPadding { get { return true; } }
        private void HitInfo_Load(object sender, System.EventArgs e) {
            DevExpress.XtraVerticalGrid.Design.XViews.ConfigureDemoView(vGridControl1);
        }

        //<vGridControl1>
        private void vGridControl1_MouseMove(object sender, System.Windows.Forms.MouseEventArgs e) {
            ShowHitInfo(vGridControl1.CalcHitInfo(new Point(e.X, e.Y)));
        }

        private void ShowHitInfo(DevExpress.XtraVerticalGrid.VGridHitInfo hi) {
            lbHitTest.Text = hi.HitInfoType.ToString();
            if(hi.Row == null)
                lbRow.Text = "No row";
            else {
                if(hi.Row is DevExpress.XtraVerticalGrid.Rows.MultiEditorRow) {
                    if(hi.CellIndex != -1)
                        lbRow.Text = hi.Row.GetRowProperties(hi.CellIndex).Caption;
                }
                else
                    lbRow.Text = hi.Row.Properties.Caption;
            }
            if(hi.HitInfoType == DevExpress.XtraVerticalGrid.HitInfoTypeEnum.ValueCell) {
                if(hi.Row is DevExpress.XtraVerticalGrid.Rows.MultiEditorRow)
                    lbCellValue.Text = vGridControl1.GetCellValue(hi.Row as DevExpress.XtraVerticalGrid.Rows.MultiEditorRow, hi.RecordIndex, hi.CellIndex).ToString();
                else
                    lbCellValue.Text = vGridControl1.GetCellValue(hi.Row, hi.RecordIndex).ToString();
            }
            else
                lbCellValue.Text = " ";
        }
        //</vGridControl1>
    }
}
